File: javascript/hamming/hamming.js
var Hamming = function() {
};

Hamming.prototype.compute = function(strand1, strand2) {
  if (strand1.length === strand2.length){
    var distance = 0;
    for (var i=0; i<strand1.length; i++){
      if (strand1.charAt(i) != strand2.charAt(i)){
        distance += 1;
      }
    }
    return distance;
  } else {
    throw new Error('DNA strands must be of equal length.');
  }
};
module.exports = Hamming;

File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

const testVersion = 6

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("Cannot compare these two strings")
	}

	distance := 0

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil
}
